utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/kafka/KafkaConnector.java

```diff
@@ -77,4 +76,23 @@ public void run() {
         thread.start();
         System.out.println("Started Kafka consumer thread");
     }
+
+    // For APM testing, produce message without starting a new thread
+    public void produceMessageWithoutNewThread(String message) throws Exception {
+        KafkaTemplate<String, String> kafkaTemplate = createKafkaTemplateForProducer();
+        System.out.printf("Publishing message: %s%n", message);
+        kafkaTemplate.send(topic, message);
+    }
+
+    // For APM testing, a consume message without starting a new thread
+    public void consumeMessageWithoutNewThread() throws Exception {
+        KafkaConsumer<String, String> consumer = createKafkaConsumer();
+        consumer.subscribe(Collections.singletonList(topic));
+        ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(3));
+        if(records.isEmpty())
+            System.out.println("No record found when polling " + topic);
```

Comment:
this is inconsistent with the `startConsumingMessages` method above which uses (nearly) infinite timeout, but I feel it makes debugging easier to have a message telling us that the message couldn't be read rather than nothing.